pilot/pkg/model/context.go

```diff
@@ -267,9 +268,15 @@ func (node *Proxy) SetServiceInstances(env *Environment) error {
 }
 
 func (node *Proxy) SetWorkloadLabels(env *Environment) error {
+	// The WorkloadLabels is already parsed from Node metadata["istio.io/metadata"][labels]
+	if node.WorkloadLabels != nil {
+		return nil
+	}
+
 	l, err := env.GetProxyWorkloadLabels(node)
 	if err != nil {
 		log.Warnf("failed to get service proxy workload labels: %v, defaulting to proxy metadata", err)
+		// TODO: node.Metadata also includes all the labels, not sure whether will be removed in future
 		l = labels.Collection{node.Metadata}
```

Comment:
this code should go away. I didn't realize that we were falling back from env to node metadata.. i think we should switch from node metadata to env and then just leave the field empty..